pkg/kubectl/cmd/create.go

```diff
@@ -172,3 +181,54 @@ func createAndRefresh(info *resource.Info) error {
 	info.Refresh(obj, true)
 	return nil
 }
+
+// nameFromCommandArgs is a utility function for commands that assume the first argument is a resource name
+func nameFromCommandArgs(cmd *cobra.Command, args []string) (string, error) {
+	if len(args) == 0 {
+		return "", cmdutil.UsageError(cmd, "NAME is required")
+	}
+	return args[0], nil
+}
+
+// createSubcommandInternal is intended to be used by subcommands that generator client side objects
+func createSubcommandInternal(f *cmdutil.Factory, cmd *cobra.Command, args []string, out io.Writer, structuredGenerator kubectl.StructuredGenerator) error {
```

Comment:
> Agree on the args[0] for name bit, but disagree on the need for an object. To me, this looks the same as RunCreate except it takes the object that knows how to create a runtime.Object instead of options.

I think once you get to 8 arguments, it needs an object.  Passing the `cmd` around to deeper and deeper functions is a serious problem for composeability and reasonable testability.  That's why origin uses the `Options` structs to strongly decouple the work from the representation on the command line.